### PR TITLE
[Bugfix] Introduce getting table options interface

### PR DESCRIPTION
**Pull request by zhoulii**
### Purpose

close: #158 

add `/api/metadata/query/options` api for getting table options.

### Tests

`MetadataControllerTest#testGetOptionInfo`

### CODE REVIEW

File: paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/MetadataControllerTest.java [hunk 1]
```diff
@@ -298,4 +298,28 @@ public void testGetSnapshotInfo() throws Exception {
...
+                                        .contentType(MediaType.APPLICATION_JSON_VALUE)
+                                        .accept(MediaType.APPLICATION_JSON_VALUE))
+                        .andExpect(MockMvcResultMatchers.status().isOk())
+                        .andDo(MockMvcResultHandlers.print())
+                        .andReturn()
+                        .getResponse()
+                        .getContentAsString();
+
+        R<Void> result = ObjectMapperUtils.fromJSON(response, new TypeReference<R<Void>>() {});
+        assertEquals(200, result.getCode());
```
Comment (SteNicholas) [inline paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/MetadataControllerTest.java:323]: Add the asertions for the data of result.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/MetadataControllerTest.java (hunk 1 above)
Comment (zhoulii) [reply paimon-web-server/src/test/java/org/apache/paimon/web/server/controller/MetadataControllerTest.java:327]: done

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java [hunk 2]
```diff
@@ -171,6 +172,26 @@ public List<DataFileVO> getDataFile(MetadataDTO dto) {
...
+    public List<OptionVO> getOption(MetadataDTO dto) {
+        initEnvironment(dto, MetadataConstant.OPTIONS);
+
+        List<OptionVO> result = new LinkedList<>();
+
+        try {
+            reader.forEachRemaining(
+                    internalRow -> {
+                        OptionVO optionVo = new OptionVO();
+                        optionVo.setKey(internalRow.getString(0).toString());
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java:185]: Is there any `NullPointerException` when `internalRow.getString(0)` is null?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java [hunk 3]
```diff
@@ -171,6 +172,26 @@ public List<DataFileVO> getDataFile(MetadataDTO dto) {
...
+                        optionVo.setValue(internalRow.getString(1).toString());
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java:186]: Ditto.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java (hunk 2 above)
Comment (zhoulii) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/MetadataServiceImpl.java:185]: Thanks for the review @SteNicholas . Maybe I did not think it through thoroughly, I cannot anticipate under what circumstances the $options table would have a null value.

**Review by SteNicholas — APPROVED** [review]

**Pull request merged**